src/wp-includes/interactivity-api/class-wp-interactivity-api.php

```diff
@@ -167,10 +167,15 @@ public function print_client_interactivity_data() {
 		}
 
 		if ( ! empty( $interactivity_data ) ) {
+			$json_encode_flags = JSON_HEX_TAG | JSON_HEX_AMP | JSON_UNESCAPED_SLASHES;
+			if ( str_starts_with( get_option( 'blog_charset' ), 'UTF-' ) ) {
```

Comment:
Maybe just checking `UTF-8` is sufficient here, it's the default and likely the most common by far. It's probably not worth worrying about `UTF-16` or `UTF-32`.

Reply:
```suggestion
			if ( 'UTF-8' === get_option( 'blog_charset' ) ) {
```